x-pack/legacy/plugins/siem/index.ts

```diff
@@ -23,21 +23,41 @@ import {
   DEFAULT_INTERVAL_VALUE,
   DEFAULT_FROM,
   DEFAULT_TO,
+  DEFAULT_SIGNALS_INDEX,
+  DEFAULT_SIGNALS_INDEX_KEY,
 } from './common/constants';
 import { defaultIndexPattern } from './default_index_pattern';
 
+// This is VERY TEMPORARY as we need a way to turn on alerting and actions
+// for the server without having to manually edit this file. Once alerting
+// and actions has their enabled true by default this can be removed.
+// 'alerting', 'actions' are hidden behind feature flags at the moment so if you turn
+// these on without the feature flags turned on then Kibana will crash since we are a legacy plugin
+// and legacy plugins cannot have optional requirements.
+// This returns ['alerting', 'actions', 'kibana', 'elasticsearch'] iff alertingFeatureEnabled is true
+// or if the developer signalsIndex is setup. Otherwise this returns ['kibana', 'elasticsearch']
+export const getRequiredPlugins = (
+  alertingFeatureEnabled: string | null | undefined,
+  signalsIndex: string | null | undefined
+) => {
+  const baseRequire = ['kibana', 'elasticsearch'];
+  if (
+    signalsIndex != null ||
```

Comment:
I am wondering if we need to check if the `signalsIndex` is not also empty.

Reply:
I can add a check to see if it is an empty string or not and not add those others if it is an empty string to be safe.